roomserver/internal/input/input_events.go

```diff
@@ -255,13 +255,32 @@ func (r *Inputer) processRoomEvent(
 				hadEvents:  map[string]bool{},
 				haveEvents: map[string]*gomatrixserverlib.HeaderedEvent{},
 			}
-			if err := missingState.processEventWithMissingState(ctx, event, headered.RoomVersion); err != nil {
+			if override, err := missingState.processEventWithMissingState(ctx, event, headered.RoomVersion); err != nil {
```

Comment:
I don't like the name `override`, it's not clear what that means. Comments are fab further down, so perhaps `stateSnapshot` or something?